clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -355,6 +355,12 @@ protected synchronized long timeToNextHeartbeat(long now) {
         // we don't need to send heartbeats
         if (state.hasNotJoinedGroup())
             return Long.MAX_VALUE;
+        if (heartbeatThread != null) {
```

Comment:
We could 1 line it ya?
```
if (heartbeatThread != null && heartbeatThread.hasFailed()) {
...
}
```

Reply:
I'm also curious, do we actually need to perform null check for heartbeatThread? We will always create one when we try or already be in a group no?

Reply:
@philipnee  For a KafkaConsumer with group.id set and **assign** mode used. The coordinator will be created, but the heartbeat thread will not be created. So the non-null judgment here makes sense.

Reply:
> We could 1 line it ya?
if (heartbeatThread != null && heartbeatThread.hasFailed()) {
...
}


Could you address this comment?

Reply:
update